openshift-kube-apiserver/admission/customresourcevalidation/node/validate_node_config.go

```diff
@@ -57,7 +76,13 @@ func toConfigNodeV1(uncastObj runtime.Object) (*configv1.Node, field.ErrorList)
 	return obj, nil
 }
 
-type configNodeV1 struct{}
+type configNodeV1 struct {
+	admission.ValidationInterface
+
+	nodeLister                   func() corev1listers.NodeLister
+	waitForReady                 func() func() bool
```

Comment:
I would prefer this to be `waitForNodeInformerSyncedFn  func() bool`

Reply:
Mhm, I think the content of the function changes, but the "pointer" to the function remains the same, so we should be fine, no ? . 

In general, the plugins are initialized, validated, and only then wired to handlers.

Maybe we could write a small program that would demonstrate the issue ?

Reply:
okay you seem to be right, I was worried it wasn't saved as a pointer https://go.dev/play/p/h_vI248v25o

Reply:
hmm, I think that our scenario is more like https://go.dev/play/p/fHVBh-U_Gtk, no ?